File: string_rotation.py
def string_rotation(str1):
	tmp = str1 + str1
	n = len(str1)
	for i in range(1,n+1):
		substring = tmp[i:]
		if(substring == str1):
			return i

a = "geeks"
res = string_rotation(a)
print(res)

File: basic_prob/string_rotation.cpp
#include<bits/stdc++.h>
using namespace std;

int rotation_count(string str)
{
	string tmp = str + str;
	int n = str.length();


	for( int i = 1; i<=n; i++)
	{
		string substring = tmp.substr(i, str.size());

		if(str == substring){
			
			return i;
		}
	}
	
	
	
}

int main()
{
	string str = "geeks";
	int res = rotation_count(str);
	cout<<"number of rotations"<<" "<<res;
	return 0;


}